### PR TITLE
feat(react): react devtool

### PR DESCRIPTION
**Pull request by danilowoz**
This introduces support for React debugging, which allows you to inspect the React component hierarchies in the iframe. This is useful for `props` debugging and understanding the component tree.

Screenshot 2021-12-03 at 19 20 23

Depends on https://github.com/codesandbox/codesandbox-client/pull/6337

### CODE REVIEW

**Comment by danilowoz** [comment]
@bvaughn, made some progress :D

**Comment by bvaughn** [comment]
Ooh this is exciting! Eager to see this when it's done!

**Comment by bvaughn** [comment]
@danilowoz Exciting to see so many updates! Let me know when it's ready for a test drive.

File: website/docs/docs/advanced-usage/components.md [hunk 1]
```diff
@@ -173,7 +173,33 @@ For situations when you strictly want to show some code and run it in the browse
   </SandpackLayout>
 </SandpackProvider>
 
-## UnstyledOpenInCodeSandboxButton & OpenInCodeSandboxButton
+## ReactDevTools
+
+Sandpack also provides a component to add a plugin for React debugging, which allows you to inspect the React component hierarchies in the iframe. This is useful for `props` debugging and understanding the component tree.
```
Comment (sannek) [inline website/docs/docs/advanced-usage/components.md:178]:
```suggestion
Sandpack also provides a component that adds a plugin for React debugging, allowing you to inspect the React component hierarchies in the iframe. This is useful for `props` debugging and understanding the component tree.
```

**Review by sannek — APPROVED** [review]

File: website/docs/docs/advanced-usage/components.md (hunk 1 above)
Comment (sannek) [reply website/docs/docs/advanced-usage/components.md:178]: (I'm not 100% sure if "instance" is the right word there at the end??)

File: sandpack-react/src/common/Layout.tsx [hunk 2]
```diff
@@ -15,13 +16,14 @@ export interface SandpackLayoutProps {
...
   theme,
+  style,
 }) => {
   const { sandpack } = useSandpack();
   const c = useClasser("sp");
 
   return (
     <SandpackThemeProvider theme={theme}>
-      <div ref={sandpack.lazyAnchorRef} className={c("layout")}>
+      <div ref={sandpack.lazyAnchorRef} className={c("layout")} style={style}>
```
[the remark this answers is not in the document]
Comment (danilowoz) [reply sandpack-react/src/common/Layout.tsx]: I didn't find a way to override styles in the component level easily, so I realized it was not passing the style props down to the component. I think we could just spread all props into the component and not only explicit pass `style`, it seems cleaver.

File: website/docs/docs/advanced-usage/components.md [hunk 3]
```diff
@@ -173,7 +173,41 @@ For situations when you strictly want to show some code and run it in the browse
...
+export default function CustomSandpack() {
+  return (
+    <SandpackProvider template="react">
+      <SandpackLayout>
+        <SandpackPreview />
+        <SandpackReactDevTools style={{ width: "50%" }} />
+      </SandpackLayout>
+    </SandpackProvider>
+  )
+}`
```
Comment (bvaughn) [inline website/docs/docs/advanced-usage/components.md:204]: Looks like you forgot the "```" ticks for formatting here.

**Pull request merged**